compiler/yaml/src/main/core/com/kitakkun/backintime/compiler/yaml/CallableSignature.kt

```diff
@@ -0,0 +1,120 @@
+        val name: String
+        val valueParameters: ParametersSignature
+
+        data class Member(
+            override val name: String,
+            override val valueParameters: ParametersSignature,
+        ) : NamedFunction
+
+        data class TopLevel(
+            val receiverClassId: String,
+            val packageFqName: String,
+            override val name: String,
+            override val valueParameters: ParametersSignature,
+        ) : NamedFunction
+    }
+}
+
+private class CallableSignatureSerializer : KSerializer<CallableSignature> {
+    override val descriptor: SerialDescriptor = PrimitiveSerialDescriptor(serialName = "CallableSignature", kind = PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): CallableSignature {
+        val value = decoder.decodeString()
+
+        return if (value.startsWith("<get-") && value.endsWith(">")) {
+            val propertyName = value.removePrefix("<get-").removeSuffix(">")
+            CallableSignature.PropertyAccessor.Getter(propertyName)
+        } else if (value.startsWith("<set-") && value.endsWith(">")) {
+            val propertyName = value.removePrefix("<set-").removeSuffix(">")
+            CallableSignature.PropertyAccessor.Setter(propertyName)
+        } else if (value == "<this>") {
+            CallableSignature.This
+        } else {
+            // com/example/ReceiverClass com/example/hoge(*, kotlin/Int)
+            // -> com/example/ReceiverClass com/example/hoge
+            val signature = value.takeWhile { it != '(' }
+            val receiverAndFunction = signature.split(" ")
+
+            val receiverPart: String
+            val functionPart: String
+            when (receiverAndFunction.size) {
+                1 -> {
+                    receiverPart = ""
+                    functionPart = receiverAndFunction.single()
+                }
+
+                2 -> {
+                    receiverPart = receiverAndFunction[0]
+                    functionPart = receiverAndFunction[1]
+                }
+
+                else -> error("Error!")
+            }
+
+            val functionName = functionPart.takeLastWhile { it != '.' && it != '/' }
+            val functionPackage = functionPart.removeSuffix(functionName)
+                .replace("/", ".")
+                .removeSuffix(".")
+
+            val parameters = if (!value.contains("(")) {
+                ParametersSignature.Any
+            } else {
+                ParametersSignature.Specified(
+                    value.dropWhile { it != '(' }.dropLastWhile { it != ')' }
+                        .split(",")
+                        .map { it.trim() }
+                        .map {
+                            if (it == "*") TypeSignature.Any
+                            else if (it.toIntOrNull() != null) TypeSignature.Generic(it.toInt())
+                            else TypeSignature.Class(it)
+                        }
+                )
```

Comment:
I think this part of code is not tested. We need to write test for it.

Reply:
[Add test for TypeSignatures](https://github.com/kitakkun/back-in-time-plugin/pull/138/commits/e6685cbf98c798b1ff38097a3e0a5eaae279a0da)